src/d3d11/d3d11_fence.h

```diff
@@ -0,0 +1,98 @@
+#pragma once
+
+#include "../dxvk/dxvk_fence.h"
+#include "../dxvk/dxvk_gpu_query.h"
+#include "../dxvk/dxvk_context.h"
+
+#include "d3d11_device_child.h"
+
+extern "C" {
+  #include "shared_fence.h"
```

Comment:
If you really need this then do in in header file like:
```c
#ifdef __cplusplus
extern "C" {
#endif
...
#ifdef __cplusplus
}
#endif
```
But, IMHO, better to have shared_fence.cpp

Reply:
Right, not sure why I did it that way.  The reason the implementation is a .c is that it is shared with vkd3d-proton.